### DISCUSSION TITLE
Listening Stats from API not consistent, nor make sense

### DISCUSSION THREAD

**Discussion by cooper-fm** [AzuraCast/AzuraCast#7157] [General]
Hi,

I'm in the middle of creating a python script, to daily transfer the data from the AC stats API to another database where I can produce my own statistics panel (sorry AC guys, but this is where you need to step your game up 😬)

I've done the script, but I've noticed the data coming through the API is not consistent, take the following example:

Parameters:
Station ID 1
Start 2024-05-21
End 2024-05-21

The first listener of an IP address of 86.146.152.214 has stats of:
connected_on: 1716256867
connected_until: 1716315890
connected_time: 19492

**ERROR 1**
The connected_time is incorrect, and the difference between the two values is actually 59,023, not 19,492.

Let's try this again, but changing the end date of the parameter to 2024-05-22

This time, the same first listener comes through with the same IP address, has the same connected_on time, but stats have changed:
connected_until: 1716401377
connected_time: 35236

_This time, the calculated connected_time should be 144510, not 35,236_ 

**ERROR 2**
So what is the real connected time, and/or how long has the listener been connected for? I could accept different connected_times as I could have calculated the difference between the two.

I've attached a few images to show the differences, together with a spreadsheet showing that the same results are common across the many listeners, and it's not a simple anomoly.

Am I missing something drastic here, my thoughts are that:
1) The connected_until value should be static, and never change
2) The connected_time should always be the difference between _on and _until

![Screenshot 2024-05-29 at 16 19 03](https://github.com/AzuraCast/AzuraCast/assets/171165732/9647a7a6-056c-4d24-93bf-a40abf5bd5d2)
![Screenshot 2024-05-29 at 15 58 49](https://github.com/AzuraCast/AzuraCast/assets/171165732/805d6014-fb25-4b45-815a-bc61054ecd76)
![Screenshot 2024-05-29 at 15 58 18](https://github.com/AzuraCast/AzuraCast/assets/171165732/5d3cfb56-c090-40d6-820c-95c8e88e1333)

**Comment by BusterNeece** [comment]
@cooper-fm Sorry for the (very) delayed response to this. Some issues have fallen through the cracks a little.

In this case, by default we group listeners into "unique" cohorts. Within these cohorts, all of the times a single listener connected are clumped together, the "start time" is the start of the first time, the "end time" is the end of the last time, and the duration is the sum total of seconds that the user was connected during that whole duration.

To prevent this grouping and get distinct records for every single connect/disconnect event, you can pass `?unique=false` in the query parameters of the API call. This will likely more closely match what you're expecting here.
